drivers/gpio/gpio_neorv32.c

```diff
@@ -234,7 +234,7 @@ static const struct gpio_driver_api neorv32_gpio_driver_api = {
 			NULL,						\
 			&neorv32_gpio_##n##_data,			\
 			&neorv32_gpio_##n##_config,			\
-			POST_KERNEL,					\
+			PRE_KERNEL_1,					\
```

Comment:
This is the only one that I'm concerned about changing here due to the initialization order of the syscon that is used on this soc.

Reply:
@henrikbrixandersen would you rather this be left as is with `POST_KERNEL` for now, and for it to fixed in another pull request with the initialization order

Reply:
Ass you notes yourself, `PRE_KERNEL_1` is for device drivers without dependencies. The NEORV32 GPIO driver does have a dependency, namely SYSCON, so what is there to fix?

Reply:
> 

My concern here is that the default for `SYSCON_INIT_PRIORITY` seems high was just arbitrarily chosen at that time. but... since this is a dependency upon something that is within `PRE_KERNEL_1`... This should rather be `PRE_KERNEL_2` based on it's definition.